lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -604,12 +667,61 @@ def cmd_hosts(*args)
     print_status("Deleted #{delete_count} hosts") if delete_count > 0
   end
 
-  def cmd_services_help
-    # Like cmd_hosts, use "-h" instead of recreating the column list
-    # here
-    cmd_services("-h")
+  #
+  # Tab completion for the services command
+  #
+  # @param str [String] the string currently being typed before tab was hit
+  # @param words [Array<String>] the previously completed words on the command line.  words is always
+  # at least 1 when tab completion has reached this stage since the command itself has been completed
+
+  def cmd_services_tabs(str, words)
+    if words.length == 1
+      return @@services_opts.option_keys.select { |opt| opt.start_with?(str) }
+    end
+
+    case words[-1]
+    when '-c', '--column'
+      # Should we get some default columns?
+      return ['created_at', 'info', 'name', 'port', '...']
+    when '-O', '--order'
+      # Return some sample column ids
+      return ['1', '2', '3', '4', '...']
+    when '-p', '--port'
+      # Might be a good idea to limit this to only a few services
+      return framework.db.services[0..4].map(&:port).uniq + ['...']
```

Comment:
```suggestion
      return framework.db.services[0..4].map(&:port).uniq
```

Reply:
I wonder if we should just return `[]` here, or check if there's prior art here? 🤔

Reply:
I've decided to return [] for now 👍